google/resource_kms_crypto_key_test.go

```diff
@@ -0,0 +1,373 @@
+	cryptoKeyName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleKmsCryptoKeyWasRemovedFromState("google_kms_crypto_key.crypto_key"),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleKmsCryptoKey_rotation(projectId, projectOrg, projectBillingAccount, keyRingName, cryptoKeyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsCryptoKeyExists("google_kms_crypto_key.crypto_key"),
+					testAccCheckGoogleKmsCryptoKeyHasRotationParams("google_kms_crypto_key.crypto_key"),
+				),
+			},
+			resource.TestStep{
+				Config: testGoogleKmsCryptoKey_removed(projectId, projectOrg, projectBillingAccount, keyRingName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsCryptoKeyWasRemovedFromState("google_kms_crypto_key.crypto_key"),
+					testAccCheckGoogleKmsCryptoKeyVersionsDestroyed(projectId, location, keyRingName, cryptoKeyName),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleKmsCryptoKeyExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		keyRingId, err := parseKmsKeyRingId(rs.Primary.Attributes["key_ring"], config)
+
+		if err != nil {
+			return err
+		}
+
+		cryptoKeyId := &kmsCryptoKeyId{
+			KeyRingId: *keyRingId,
+			Name:      rs.Primary.Attributes["name"],
+		}
+
+		listCryptoKeysResponse, err := config.clientKms.Projects.Locations.KeyRings.CryptoKeys.List(cryptoKeyId.parentId()).Do()
+		if err != nil {
+			return fmt.Errorf("Error listing KeyRings: %s", err)
+		}
+
+		for _, cryptoKey := range listCryptoKeysResponse.CryptoKeys {
+			log.Printf("[DEBUG] Found CryptoKey: %s", cryptoKey.Name)
+
+			if cryptoKey.Name == cryptoKeyId.cryptoKeyId() {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("CryptoKey not found: %s", cryptoKeyId.cryptoKeyId())
+	}
+}
+
+func testAccCheckGoogleKmsCryptoKeyHasRotationParams(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		keyRingId, err := parseKmsKeyRingId(rs.Primary.Attributes["key_ring"], config)
+
+		if err != nil {
+			return err
+		}
+
+		cryptoKeyId := &kmsCryptoKeyId{
+			KeyRingId: *keyRingId,
+			Name:      rs.Primary.Attributes["name"],
+		}
+
+		getCryptoKeyResponse, err := config.clientKms.Projects.Locations.KeyRings.CryptoKeys.Get(cryptoKeyId.cryptoKeyId()).Do()
+
+		if err != nil {
+			return err
+		}
+
+		_, err = time.Parse(time.RFC3339Nano, getCryptoKeyResponse.NextRotationTime)
+
```

Comment:
I think you should also make sure `getCryptoKeyResponse.RotationPeriod` exists.  CryptoKeys are allowed to have a `NextRotationTime` without having a `RotationPeriod`, but that would be bad in this case, since you'd be creating it with both fields.